Userland/Applications/FileManager/PropertiesWindow.cpp

```diff
@@ -234,6 +241,39 @@ ErrorOr<void> PropertiesWindow::create_file_type_specific_tabs(GUI::TabWidget& t
     return {};
 }
 
+ErrorOr<void> PropertiesWindow::create_archive_tab(GUI::TabWidget& tab_widget, NonnullRefPtr<Core::MappedFile> mapped_file)
+{
+    auto maybe_zip = Archive::Zip::try_create(mapped_file->bytes());
+    if (!maybe_zip.has_value()) {
+        warnln("Failed to read zip file '{}' ", m_path);
+        return {};
+    }
+    auto zip = maybe_zip.release_value();
+
+    auto tab = TRY(tab_widget.try_add_tab<GUI::Widget>(TRY("Archive"_string)));
+    TRY(tab->load_from_gml(properties_window_archive_tab_gml));
+
+    size_t file_count = 0;
```

Comment:
That looks way too similar to what we have in `file.cpp` :eyes: 
What about putting that in a method of `Archive::Zip`?

Reply:
Yep, this same code is appearing in `file.cpp`, `unzip.cpp`, and now here. I think it makes sense to add an API to `LibArchive` to get this information. Doesn't have to be in this PR though, necessarily.